cmake/SpectreCheckDependencies.cmake

```diff
@@ -22,6 +22,8 @@ function(check_spectre_libs_dependencies)
       ${SPECTRE_LIBS}
       ALLOWED_EXTRA_TARGETS
       SpectreFlags
+      SpectreAllocator
```

Comment:
Is order important?  If not, alphabetical?

Reply:
I've alphabetized. I also removed Jemalloc because the recursive change that Nils suggested made it unnecessary.